ibis/backends/sql/compilers/datafusion.py

```diff
@@ -107,7 +115,7 @@ def visit_NonNullLiteral(self, op, *, value, dtype):
                 unit=sg.exp.var(dtype.unit.plural.lower()),
             )
         elif dtype.is_timestamp():
-            return self._to_timestamp(value, dtype, literal=True)
+            return self._to_timestamp(str(value), dtype, from_=op.dtype)
```

Comment:
Why does the value need to be converted to a string here?

Reply:
hm, It might not be. I was just porting over the logic that existed before (the `str(value)` was INSIDE the _to_timestamp() function before). I'll remove it and see if it works.

I think one thing to think about is if we want to support both `ibis.literal("2025-04-17", "timestamp")` as well as `ibis.literal(123456.7, "timestamp")`. I think we probably only currently test for the string version, and I think it is probably fine to leave the floating version as undefined behavior that we don't test for.

Reply:
I removed the str() cast here. I also changed this to `from_=dt.infer(value)` so that _to_timestamp() casts to the right precision of timestamp for `int`s